documentation/polymer-templates/tutorial-template-intro.asciidoc

```diff
@@ -1,59 +1,44 @@
 
-== When to Use Java Components
 
-The traditional way of building UIs in Vaadin is to use the Java API. This works very well for complex and dynamic UIs where the view structure changes depending on the data that is viewed.
+For example `<vaadin-button id="my-button">Press me</vaadin-button>` can be accessed with `@Id("my-button") Button button;` in Java.
 
-The benefits of using the Java API include:
+As declarative layouts are a robust and maintainable way for defining views, https://vaadin.com/designer[Vaadin Designer], the visual tool to build UI for Vaadin apps, uses them as its format.
 
-* `Component` API: This allows you to use inheritance to easily create and modify your own custom components. 
-* `Element` API: This is a lower-level API that gives you more control over the DOM than ever before. 
-* All Java code is completely type-safe: The risk of typos or other inadvertant mistakes is greatly reduced because the code is compiled before use. 
+== Known Limitations of Templates
 
-The disadvantage of this approach is that it is harder to get a structural overview of the resulting view from the code. In addition, although modifying the DOM is possible, it is not as easy as simply writing HTML in a template. 
+The template API has limitations that you need to be aware of before using it.
 
+* Currently HTML templates are not processed on the server before being sent to the client. Server side is not aware of the content of the template, like the attributes used by a component. This means that many getter methods return the incorrect value because the client never sent them to the server. We plan to remove this limitation in the near future.
+* You can add components and a DOM to a template structure, but you cannot remove anything present in the original template. However, overriding component properties do work. 
 
-== Known Limitations of Polymer Templates
+== Correctly Separating Structure and Logic
+Strive to only define things that do not change in the template and leave all dynamic parts to Java. In the template you can, for example, have attributes that never change, captions for components like `Button` and `TextField` that always stay the same, or styling for the view. If, for example, a `Button` should change the caption depending on user interaction, then leave it empty in the template and define the intial value in Java.
 
-The `PolymerTemplate` API has intentional limitations that you need to be aware of before chosing to use it. These limitations are by design. The root cause of the limitations is that HTML templates are not processed on the server before being sent to the client. Because templates are not checked on the server:
+With `Grid`, defining columns can feel like a static part, but columns and data go strongly hand-in-hand. We recommend defining a `vaadin-grid` with static attributes in the template and defining columns and data in Java.
```

Comment:
For consistency with above pattern of no hyphenation when the composite is a proper noun (cf.  "future proof") this should probably be "[...] go strongly **hand in hand**.".

Reply:
Done.